### DISCUSSION TITLE
in non-react context，how to get the value of atom ?

### DISCUSSION THREAD

**Discussion by WangYuLue** [pmndrs/jotai#1028] [Q&A] [answered]

**Comment by dai-shi** [comment, marked as answer]
jotai atoms are modeled after `useState`, so in general, it can exists only in a React component tree.

That said, there are three approaches:

## `useEffect`

This is the same as `useState` pattern.
```js
let globalValue

  // in component/hook
  const [value] = useAtom(valueAtom)
  useEffect(() => {
    globalValue = value
  })
```

## use "global" state library integrations

atoms are not truly global, so integrate with global (module) state using either `jotai/valtio`, `jotai/zustand` or `jotai/redux`.

## experimental `unstable_createStore`

We are experimenting a new feature. See: #861 and #922

**Comment by WangYuLue** [comment]
thanks very much
